src/compare/compare.py

```diff
@@ -0,0 +1,84 @@
+import os
+import sys
+import time
+import configparser
+import numpy as np
+import mmh3
+import itertools
+import pickle
+import redis
+import psycopg2
+from termcolor import colored
+
+from pyspark.conf import SparkConf
+from pyspark.context import SparkContext
+from pyspark.sql import SQLContext
+from pyspark.sql.functions import udf, col
+
+from pyspark.sql.types import IntegerType, FloatType, ArrayType
+
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))) + "/config")
+#sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))) + "/lib")
+import config
+#import util
+
+def compare_text():
+
+    cf = configparser.ConfigParser()
+    cf.read('../config/db_properties.ini')
+    print(colored("Reading shingle data from redis", "green"))
+    
+    print(colored("Calculating minhash value overlap for articles in each category and writing results out to database", "green"))
+  
+    # Set up postgres connection for writing similarity scores 
+    connection = psycopg2.connect(host=cf['postgres']['url_results'], database='similarity_scores', user=cf['postgres']['user'], password=cf['postgres']['password'])
+    cursor = connection.cursor()    
+
+    # Set up redis connection for reading in minhash values
+    rdb = redis.StrictRedis(config.REDIS_SERVER, port=6379, db=0)
+
+    # For each category, go through each pair of articles and write the ones with a high enough minhash overlap to a database
+    URL_HEADER = 'https://www.wikipedia.org/wiki?curid='
+    for category in rdb.scan_iter('cat:*'):
+        pairs = list(itertools.combinations(eval(list(rdb.smembers(category))[0]), 2))
+        print("Evaluating potential for {} pairs in category {}".format(len(pairs), category))
+        for pair in pairs:
+           minhash1 = rdb.smembers('id:{}'.format(pair[0]))
+           minhash2 = rdb.smembers('id:{}'.format(pair[1]))
+           if minhash1 and minhash2:
+               minhash1 = eval(list(minhash1)[0])
```

Comment:
It is best practice to not use eval(). It can be used to compromise your python script. Image someone spoofed the redis instance and made it return whatever they want. Instead it would be better to make a `deserialize_redis_set()` function, that takes in the redis response and tries to safely construct the list/set.

Reply:
@tyaq good point. There's also a function literal_eval which is similar to eval in that it parses a string but doesn't have security risks since it doesn't evaluate code. Using just list() functions breaks up the text in the wrong way, so I'll stick with this now